clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -261,6 +261,32 @@ public void testFencingWhenStateIsPrepareLeaving() {
         assertEquals(MemberState.LEAVING, membershipManager.state());
     }
 
+    @Test
+    public void testAssignmentReceivedWhenStateIsPrepareLeaving() {
```

Comment:
Could you have a test for each state that cannot handle new assignments? I guess that the states that can handle new assignments are tested elsewhere.

Reply:
Currently `PREPARE_LEAVING` is the only state where this could happen. It's the only one where a transition to RECONCILE is not valid, but the member is still sending HB (meaning that the broker could send a new assignment anytime). All other states either allow the transition to RECONCILE (tested in multiple other tests), or do not send HB (ex. fatal). I preferred to add the check as a generic one based on the `previousValidStates` in case some new states in this situation comes out (as this is still evolving quite a bit)